// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";
option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.segmentation_platform.proto";
option java_outer_classname = "SegmentationProto";

package segmentation_platform.proto;

// List of user segment types.
// Any segment that needs a server provided model should have an entry in
// OptimizationTarget. If the segment only uses default model, then the segment
// should have a value higher than `MAX_OPTIMIZATION_TARGET`.
// Keep in sync with SegmentationPlatformSegmentationModel in
// //tools/metrics/histograms/enums.xml.
enum SegmentId {
  // Reserved fields are `OptimizationTarget`s that are not usde by segmentation
  // to prevent accidental misuse.
  reserved 1, 2, 3, 7, 8, 9, 13, 14, 15, 19, 20, 24, 25, 26, 30, 31;

  OPTIMIZATION_TARGET_UNKNOWN = 0;
  // Target for segmentation: New tab page user.
  OPTIMIZATION_TARGET_SEGMENTATION_NEW_TAB = 4;
  // Target for segmentation: Share user.
  OPTIMIZATION_TARGET_SEGMENTATION_SHARE = 5;
  // Target for segmentation: Voice user.
  OPTIMIZATION_TARGET_SEGMENTATION_VOICE = 6;
  // Target that enables data collection on client side for various experiments.
  OPTIMIZATION_TARGET_SEGMENTATION_DUMMY = 10;
  // Target for segmentation: Chrome Android Start user.
  OPTIMIZATION_TARGET_SEGMENTATION_CHROME_START_ANDROID = 11;
  // Target for segmentation: Query Tiles user.
  OPTIMIZATION_TARGET_SEGMENTATION_QUERY_TILES = 12;
  // Target for segmentation: Determine users with low engagement with chrome.
  OPTIMIZATION_TARGET_SEGMENTATION_CHROME_LOW_USER_ENGAGEMENT = 16;
  // Target for segmentation: Determine users who prefer to use Feed.
  OPTIMIZATION_TARGET_SEGMENTATION_FEED_USER = 17;
  // Target for price tracking action when shown as a contextual page action.
  OPTIMIZATION_TARGET_CONTEXTUAL_PAGE_ACTION_PRICE_TRACKING = 18;
  // Target for segmentation: Determine users who are interested in shopping.
  OPTIMIZATION_TARGET_SEGMENTATION_SHOPPING_USER = 21;
  // Target for segmentation: Chrome Android Start user V2.
  OPTIMIZATION_TARGET_SEGMENTATION_CHROME_START_ANDROID_V2 = 22;
  // Target for segmentation: Determine users who frequently search.
  OPTIMIZATION_TARGET_SEGMENTATION_SEARCH_USER = 23;
  // Target for segmentation: Segment of users who switched devices.
  OPTIMIZATION_TARGET_SEGMENTATION_DEVICE_SWITCHER = 27;
  // Target for segmentation: Adaptive toolbar.
  OPTIMIZATION_TARGET_SEGMENTATION_ADAPTIVE_TOOLBAR = 28;
  // Target for segmentation: Determine users who are tabletproductivity users.
  OPTIMIZATION_TARGET_SEGMENTATION_TABLET_PRODUCTIVITY_USER = 29;
  // Target for web app install promotion.
  OPTIMIZATION_TARGET_WEB_APP_INSTALLATION_PROMO = 32;
  // Add new entries to OptimizationTarget proto.

  // New entries should start from a 1000 if OptimizationTarget does not
  // have a corresponding type.
  MAX_OPTIMIZATION_TARGET = 999;

  // Determine power users of the browser app.
  POWER_USER_SEGMENT = 1000;
  // Determine users active on multiple synced devices.
  CROSS_DEVICE_USER_SEGMENT = 1001;
  // User segment that regularly use chrome features.
  FREQUENT_FEATURE_USER_SEGMENT = 1002;
  // Target for intentional users (people opening Chrome on their own instead of
  // arriving from other apps).
  INTENTIONAL_USER_SEGMENT = 1003;
  // User segment: User is more likely to use resume features in chrome..
  RESUME_HEAVY_USER_SEGMENT = 1004;
  // Determine device tiers for different user devices.
  DEVICE_TIER_SEGMENT = 1005;
};
